Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/ContestsController.cs

```diff
@@ -592,13 +598,23 @@ public ActionResult CalculateContestLoad(ContestLoadCalculationViewModel model)
             {
                 var contest = this.contestsData.GetById(model.PreviousContestId.Value);
                 model.PreviousContestSubmissions = this.GetOfficialSubmissionsByContest(contest.Id);
-                model.PreviousContestExpectedProblems = contest.ProblemGroups.SelectMany(x => x.Problems).Count();
+                model.PreviousContestExpectedProblems = contest.ProblemGroups.Where(pg => !pg.IsDeleted).Count();
                 model.PreviousContestParticipants = contest.Participants
                     .Where(x => x.IsOfficial == true)
                     .Count();
-                model.PreviousAverageProblemRunTimeInSeconds = 
+                model.PreviousAverageProblemRunTimeInSeconds =
                     this.contestsBusiness.GetContestSubmissionsAverageRunTimeSeconds(contest);
             }
+            else
+            {
+                if (model.PreviousAverageProblemRunTimeInSeconds == null
+                    || model.PreviousContestParticipants == null
+                    || model.PreviousContestExpectedProblems == null
+                    || model.PreviousContestSubmissions == null)
```

Comment:
Why not extract this into `model.PreviousContest` and have all the data there. Here you will just check if PreviousContest is null. Also these properties will not have to be nullable inside.